### PR TITLE
feat: adding start and end year date pickers to the form 

### PR DESCRIPTION
**Pull request by shivamksharma**
using https://mui.com/x/react-date-pickers/date-picker/

- Adding DatePicker components for selecting start and end years in the degree planner form 
- Styled the DatePicker inputs
- Ensured a consistent user interface with updated CSS 
- The layout is still good, and the new date pickers fit in nicely.

Closes: #26

### CODE REVIEW

**Review by GoodGameRuler — COMMENTED** [review]
Generally, I wouldn't approve drastic styling changes, but we don't have a style guide in place yet. Will raise an issue for this. Tagging @jennifermtan. Otherwise good stuff.

**Review by GoodGameRuler — CHANGES_REQUESTED** [review]
I will request that you change the CSS of the date pickers to match the other selection inputs.

image


Also, please revert the MUI button to the original HTML one, unless you can justify including this would benefit the codebase.

**Comment by GoodGameRuler** [comment]
> > I will request that you change the CSS of the date pickers to match the other selection inputs.
> > image
> > Also, please revert the MUI button to the original HTML one, unless you can justify including this would benefit the codebase.
> 
> I was the one who suggested to use the MUI date pickers because I had that in the prototype + it helps choosing dates in a calendar format already. Let me know if you think otherwise! But I agree with consistent styling for this :)

Nono, the submit button was also change to an mui button. Wasn't referring to the date pickers.

**Review by jennifermtan — CHANGES_REQUESTED** [review]
Please address the following changes and adjust css styling to be consistent with the current UI as shown in #26.

File: app/degreePlanner/page.jsx [hunk 1]
```diff
@@ -1,63 +1,132 @@
...
+                  Select minor
+                </option>
+                <option value="1">Mathematics</option>
+                <option value="2">Computer Science</option>
+                <option value="3">Design</option>
+                <option value="4">NA</option>
+              </select>
+            </div>
+            <div className="formElement">
+              <Button
```
[the remark this answers is not in the document]
Comment (shivamksharma) [reply app/degreePlanner/page.jsx:116]: okay, I'll fix it  !

File: app/degreePlanner/detailPlanner.css [hunk 2]
```diff
@@ -1,166 +1,181 @@
 .degreePlanContainer {
-    display: flex;
-    flex-direction: row;
-    width: 100%;
-    align-items: space-between;
-
+  display: flex;
```
Comment (jennifermtan) [inline app/degreePlanner/detailPlanner.css:2]: Please don't edit any css that you didn't touch!

**Comment by jennifermtan** [comment]
I don't see any changes in the date picker CSS in the preview as shown below:
Screenshot 2024-05-31 at 2 36 23 PM

Also, please pull from main as there have been CSS changes for it. Check what the form in https://www.unitrack.au/degreePlanner looks like!

**Comment by GoodGameRuler** [comment]
Marking this duplicate and closing this PR refer to #56.

**Comment by GoodGameRuler** [comment]
Marking this duplicate and closing this PR refer to #56.

File: app/degreePlanner/detailPlanner.css (hunk 2 above)
Comment (shivamksharma) [reply app/degreePlanner/detailPlanner.css:2]: it indentation issue, nothing else !

**Pull request closed without merging**